obsolete/weather/weather/wm918/wx200d-1.2/wxfilter.1
.\" @(#)$Id: wxfilter.1,v 1.1 2002/09/23 19:12:51 bogdan Rel $
.\"
.\" Copyright (C) 1998 - 2000 Tim Witham <twitham@quiknet.com>
.\"
.\" (see the files README and COPYING for more details)

.TH WXFILTER 1 "Feb 01, 2000" "wx200d Manual"
.SH NAME
wxfilter \- WX200 weather data filter
.SH SYNOPSIS
.B wxfilter
[-h] [-z]

.SH DESCRIPTION

.B wxfilter
takes binary WX200 data on the standard input and produces
tab-delimited ASCII or abbreviated binary on the standard output.  See
.B wx200d(5)
for a description of these file formats.
.B wxfilter
can be used with
.B wxstdout(1)
to convert real-time WX200 data, or with
.B cat(1)
or
.B zcat(1)
to convert binary
.B wx200d(8)
data files.  See
.I EXAMPLES
below.

.SH OPTIONS

.TP 0.5i
.B -h
Display a help usage message showing these options, then exit.

.TP 0.5i
.B -z
Produce abbreviated binary output, instead of tab-delimited ASCII.
This removes WX200 data groups in which no bits have changed,
resulting in the smallest file that still contains all information.
Using this option on a binary file collected by
.B wx200d(8)
will have no effect since
.B wx200d(8)
writes files in abbreviated format directly.  This option would only
be useful to convert raw data collected from a physical WX200.
Without this option, tab-delimited ASCII is produced.

.SH EXAMPLES

.TP 0.5i
.B wxstdout | wxfilter
Monitor the current WX200 data in tab-delimited format.

.TP 0.5i
.B zcat 19980206.bin.gz | wxfilter > 19980206.tab
Convert a binary
.B wx200d(8)
log file to a tab-delimited log.

.SH LIMITATIONS

.SH BUGS
I don't think so.

.SH SEE ALSO
wxstdout(1), wxdebug(1), wx200d(5), wx200d(8)

.SH AUTHOR
Tim Witham <twitham@quiknet.com>
.SH COPYRIGHT
Copyright (C) 1998 - 2000 Tim Witham <twitham@quiknet.com>

wxfilter is released under the conditions of the GNU General Public
License.  See the files README and COPYING in the distribution for
details.


sip_router/NEWS
XXX content of this file reflects a future release;
    features described here may or may not be available
    in the CVS tree
    

Release notes for SIP Express Router (ser)
***********************************************

$Id: NEWS,v 1.12 2003/04/18 14:11:30 jiri Exp $

***********************************************
* Changes introduced in 0.8.11
***********************************************

+--------------------------------------------------------+
| CAUTION: the 0.8.11 release include changes which      |
| are incompatible with scripts and databases used       |
| in previous versions. Care is advised when upgrading   |
| from previous releases to 0.8.11.                      |
+--------------------------------------------------------+

New features
=============
- RFC3261 support
	- TCP support and cross-transport forwarding [core]
	- loose routing support [rr module]
- New modules
	- vm -- voicemail interface [vm]
	- ENUM support [enum]
	- presence agent [pa]
	- dynamic domain management -- allows to manipulate 
  	  hosting of multiple domains in run-time [module]
	- flat-text-file database support [dbtext]
	- rich access control lists [permissions]
- Feature Improvements
	- click-to-dial, which is based on improved tm/FIFO 
	  that better supports external applications [tm module]
	- web accounting -- acc module can report to serweb
  	  on placed calls [acc module]
	- improved exec module (header fields passed now
      as environment variables to scripts) [exec module]
- Architectural Improvements
	- powerpc fast locking support
	- netbsd support
	- 64 bits arch. support (e.g. netbsd/sparc64).
- New Experimental Features (not tested at all yet)
	- nathelper utility for Cisco/ATA NAT traversal [nathelper]
	- another NAT traversal utility [mangler]
	- postgress support [postgress]
	- fcp module [fcp]
	- pdt module (prefix2domain) [pdt]

Changes to use of ser scripts
=============================

About Multiple Transport Support
--------------------------------
SER now suports multiple transport protocols: UDP and TCP. As there
may be UAs which support only either protocol and cannot speak to
each other directly, we recommend to alway record-route SIP requests,
to keep the transport-translating SER in path. Also, if a destination
transport is not known, stateful forwarding is recommended -- use of
stateless forwarding for TCP2UDP would result in loss of reliability.


core
----
- reply_route has been renamed to failure_route -- the old name caused
  too much confusion
- forward_tcp and forward_udp can force SER to forward via specific
  transport protocol

acc module:
-----------
- radius and sql support integrated in this module; you need to
  recompile to enable it
- acc_flag is now called log_flag to better reflect it relates
  to the syslog mode (as opposed to sql/radius); for the same
  reasons, the accounting action is now called "acc_log_request"
- log_fmt allows now to specify what will be printed to syslog

auth module:
------------
- auth module has been split in auth, auth_db, auth_radius, group
  group_radius, uri and uri_radius 
- all the parameters that were part of former auth module are now 
  part of auth_db module
XXX 

im module:
----------
- im is no longer used and has been obsoleted by TM

exec module:
------------
- exec_uri and exec_user have been obsoleted by exec_dset; 
  exec_dset is identical to exec_uri in capabilities; it 
  additionaly passes content of request elements (header 
  fields and URI parts) in environment variables; users of 
  exec_user can use exec_dset now and use the "URI_USER"  
  variable to learn user part of URI
- exec_dset and exec_msg return false, if return value of 
  script does not euqal zero
- exec_dset takes an additional parameter, which enables 
  validation of SIP URIs returned by external application

jabber module:
--------------
- presence support for Jabber users is enabled loading the PA
  module and using subscribe("jabber") for SUBSCRIBE requests to
  jabber user 

msilo module:
-------------
- m_store has now a parameter to set what should be considered
  for storing as destination uri. This enables support for saving
  the messages on negative replies.

radius_acc module:
------------------
- radius_acc module has been removed and radius accounting 
  is now part of acc module

registrar/usrloc modules:
-------------------------
XXX
  
rr module:
----------
- addRecordRoute has been replaced with record_route
- rewriteFromRoute has been replaced with loose_route()
- a new option, "enable_full_lr" can be set to make life
  with misimplemented UAs easier and put LR in from "lr=on"

tm module:
----------
- t_reply_unsafe, used in former versions within reply_routes,
  is deprecated; now t_reply is used from any places in script
- t_on_negative is renamed to t_on_failure -- the old name just
  caused too much confusion
- FIFO t_uac used by some applications (like serweb) has been
  replaced with t_uac_dlg (which allows easier use by dialog-
  oriented applications, like click-to-dial) 
- if you wish to do forward to another destination from 
  failure_route (reply_route formerly), you need to call t_relay
  or t_relay_to explicitely now
- t_relay_to has been replaced with t_relay_to_udp and t_relay_to_tcp



obsolete/weather/weather/wm918/wx200d-1.2/wxfilter.c
/*
 * @(#)$Id: wxfilter.c,v 1.1 2002/09/23 19:12:51 bogdan Rel $
 *
 * Copyright (C) 1998 - 2000 Tim Witham <twitham@quiknet.com>
 *
 * (see the files README and COPYING for more details)
 *
 * This file implements a conversion filter (stdin to stdout).
 *
 */

#include <stdio.h>
#include <unistd.h>
#include <getopt.h>
#include "wx200.h"

int
main(int argc, char **argv) {
  const char *flags = "Hhz";
  unsigned int opt, bufnum, opt_z = 0;

  while ((opt = getopt(argc, argv, flags)) != EOF) {
    switch (opt) {
    case '?':
    case 'h':
    case 'H':
      printf("usage: wxz [-h | -z]
	-h	show this help and exit
	-z	produce abbreviated binary, instead of tab-delimited ASCII\n");
      exit(0);
    case 'z':
      opt_z = 1;
    }
  }

  wx200bufinit();
  while ((bufnum = wx200bufread(0)) != EOF) {
    switch (bufnum) {
    case GROUP8:
    case GROUP9:
    case GROUPA:
    case GROUPB:
    case GROUPC:
    case GROUPF:
      if (opt_z)		/* compress */
	write(1, groupbuf[bufnum], grouplen[bufnum]);
      else if (wx200tab(bufnum)) /* tab-delimit */
	write(1, tabbuf, strlen(tabbuf));
    }
  }
  exit(0);
}
